backend/accounts/baseaccount_test.go

```diff
@@ -46,7 +47,9 @@ func TestBaseAccount(t *testing.T) {
 		GetSigningConfigurations: nil,
 		GetNotifier:              nil,
 	}
-	account := NewBaseAccount(cfg, logging.Get().WithGroup("baseaccount_test"))
+
+	coin := &mocks.CoinMock{}
```

Comment:
This would've worked as well but without any new tool dependency:

```go
type testCoin struct {
  code Code
  coinUnit, feeUnit, smallestUnit string
  coinDecimals, unitDecimals int
  urlPrefix string
}

func (tc *testCoin) Code() Code { return tc.code }
func (tc *testCoin) BlockExplorerTransactionURLPrefix() string { return tc.urlPrefix }
func (tc *testCoin) SmallestUnit() string { return tc.smallestUnit }
...
func (tc *testCoin) Initialize() {}
func (tc *testCoin) Close() error { return nil }
```

Write once and use everywhere in tests.
But I know you like mocks so won't insist.

Reply:
what you wrote is basically what the new tool generates. 

i can also just not add it to the Makefile and use it locally instead of writing it by hand, would almost come out to the same result :P

Reply:
yes, similar functionality except ~40 SLOC, no dependencies and easy to read VS ~400 SLOC, one extra tool and 10 sync.RWMutex. compare the above 1 line `Coin` method with the generated version:

```go
func (mock *CoinMock) Code() coin.Code {
	if mock.CodeFunc == nil {
		panic("CoinMock.CodeFunc: method is nil but Coin.Code was just called")
	}
	callInfo := struct {
	}{}
	lockCoinMockCode.Lock()
	mock.calls.Code = append(mock.calls.Code, callInfo)
	lockCoinMockCode.Unlock()
	return mock.CodeFunc()
}
``` 

yeah, you can say you'll never need to look at the generated source. but when a test fails, you might actually need to. plus, these giant generated methods may also introduce bugs. not today but when you or someone else `go get -u moq` next time and fetch a shiny new moq version to generate a brand new mock.

Reply:
on the flip side, if i know what the tool does, i don't have to read the code; but i'll have to read custom mock structs every time...

but sure, i can get rid of the dependency, and manually delete the mutexes, and end up with the code i would have written by hand. For example like done here: https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/backend/coins/btc/blockchain/mocks/mock.go

Reply:
> if i know what the tool does, i don't have to read the code; but i'll have to read custom mock structs every time...

well, I have to read any code every time when it isn't clear to me, including dubious function signatures like `func f() (string, string, string)` :)

but yeah, sure. I see it. good points. 

> For example like done here: https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/backend/coins/btc/blockchain/mocks/mock.go

that one actually looks much nicer. but apart from mutexes, one other difference is it doesn't panic if a stub function doesn't exist. the moq's mock, on the other hand, does.

anyway, like I said I don't mind either way. in the end of the day, it's just my personal preference.

Reply:
ok, after talking offline I like moq a bit more now :)